### PR TITLE
Move out table formatting in `osc dist` into helper function

### PR DESCRIPTION
**Pull request by abitrolly**
This is a cleanup of #863 and refactoring of internals.

* fixed typo in core `get_distributions` method name
* moved out formatting code into `util.helper.format_table` function
* made `core` method return list of dicts instead of list of strings

### CODE REVIEW

File: osc/core.py [hunk 1]
```diff
@@ -5603,11 +5603,13 @@ def get_repositories(apiurl):
     return r
 
 
-def get_distibutions(apiurl, discon=False):
-    r = []
+def get_distributions(apiurl, discon=False):
+    """Returns list of dicts with headers
+      'distribution', 'project', 'repository', 'reponame'"""
```
Comment (dmach) [inline osc/core.py:5608]: This doesn't apply to the `DISCONTINUED` distributions below.
Could you rewrite the code to return list of dicts too?

**Review by dmach — CHANGES_REQUESTED** [review]

File: osc/core.py (hunk 1 above)
Comment (abitrolly) [reply osc/core.py:5608]: The option to list discontinued distributions was commented out about 9 years ago.

https://github.com/openSUSE/osc/blame/5d7bd223400b181b061935a871faf6db9355d1d4/osc/commandline.py#L5329

So I propose to just remove this block.

Comment (dmach) [reply osc/core.py:5608]: Agreed.
Could you remove the `discon=False` argument and the whole related block of code then?

Comment (abitrolly) [reply osc/core.py:5608]: Removed and rebased.

**Review by dmach — APPROVED** [review]

**Pull request merged**